ledger/complete/ledger_stats.go

```diff
@@ -0,0 +1,46 @@
+package complete
+
+import (
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/flattener"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/node"
+)
+
+type LedgerStats struct {
+	TrieCount        uint64 `json:"tries_count"`
+	NodeCount        uint64 `json:"node_count"`
+	InterimNodeCount uint64 `json:"interim_node_count"`
+	LeafNodeCount    uint64 `json:"leaf_node_count"`
+}
+
+func (l *Ledger) CollectStats(payloadCallBack func(payload *ledger.Payload)) (*LedgerStats, error) {
+	visitedNodes := make(map[*node.Node]uint64)
+	var interimNodeCounter, leafNodeCounter, totalNodeCounter uint64
+
+	tries, err := l.Tries()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, trie := range tries {
+		for itr := flattener.NewUniqueNodeIterator(trie.RootNode(), visitedNodes); itr.Next(); {
```

Comment:
This for loop will take a long time, would we like to print the progress. Just like [what I did here](https://github.com/onflow/flow-go/blob/master/ledger/complete/wal/checkpointer.go#L448). 

We can use trie.AllocatedRegCount() to estimate the total number of nodes for reporting progress.

Do we want to use `for _, payload := range := trie.AllPayloads()`? AllPayloads runs concurrently, would be much faster.

Reply:
Since we want to look at all tries and not double count the visited nodes that's why we do it this way, otherwise for single tries we have metrics to use. (considering we fix the register fix thing).

Reply:
adding progress bar is a good idea